include/Socket.h

```diff
@@ -3,20 +3,26 @@
 
 #include <string>
 #include <unistd.h>
+#include <vector>
+
+#define MAX_BUFFER_SIZE 1024
 
 class Socket {
 public:
     Socket(int sock);
     ~Socket();
     void write(const char* buff, size_t buffsize);
     void write(const std::string& buff);
-    void writeEOL();
-    void read(char* buff, size_t buffsize);
-    size_t readLine(char* buff, size_t buffsize);
+    void read(char* buff, const size_t buffsize);
```

Comment:
const is not only redundant since size_t is passed by value, but also creates unnecessary restrictions on the caller's side.